Source/DafnyCore/Compilers/CSharp/Compiler-Csharp.cs

```diff
@@ -3435,5 +3443,30 @@ protected class ClassWriter : IClassWriter {
       TrStmt(recoveryBody, catchBlock);
     }
 
+    protected void EmitCoverageReportInstrumentation(Program program, ConcreteSyntaxTree wr) {
+      wr.WriteLine(@"
+namespace DafnyProfiling {
+  public class CodeCoverage {
```

Comment:
Could we put this in the runtime?

Reply:
Not without breaking the existing `/coverage` option, since the idea is that you could provide your own implementation of `DafnyProfilng.CodeCoverage` as per the `BranchCoverage.dfy` test.